metaflow/plugins/secrets/secrets_decorator.py

```diff
@@ -0,0 +1,228 @@
+import json
+import os
+import re
+from collections import defaultdict
+
+from metaflow.exception import MetaflowException
+from metaflow.decorators import StepDecorator
+from metaflow.plugins.secrets.aws_secrets_manager_secrets_provider import (
+    AwsSecretsManagerSecretsProvider,
+)
+from metaflow.plugins.secrets.inline_secrets_provider import InlineSecretsProvider
+
+SECRETS_BACKEND_PROVIDERS = {
+    "aws-secrets-manager": AwsSecretsManagerSecretsProvider(),
+    "_inline": InlineSecretsProvider(),
```

Comment:
We should document this somewhere that the name of the `DEFAULT_SECRETS_BACKEND_TYPE ` is `aws-secrets-manager`. Had to siff through code to dig this out.

Reply:
There is no default value for `DEFAULT_SECRETS_BACKEND_TYPE`.  When that config is unset, each secret spec must mention its backend type explicitly.

Your comment is a good reminder to enumerate the possible backend types in certain error messages though.  I will add that.